src/hooks/http.js

```diff
@@ -90,19 +93,24 @@ export const httpRequestEmitBeforeHookWrapper = (requestData, requestRandomId) =
     if (args[0] === 'socket') {
       if (isEmptyString(requestData.body)) {
         const body = extractBodyFromEmitSocketEvent(args[1]);
-        if (body) requestData.body += body;
+        if (body) {
+          requestData.body += payloadStringify(body);
+        }
       }
     }
   };
 };
 
 const createEmitResponseOnEmitBeforeHookHandler = (requestData, requestRandomId, response) => {
   let body = '';
+  const payloadSize = getEventEntitySize();
   return function(args) {
     const receivedTime = new Date().getTime();
     const { headers, statusCode } = response;
     if (args[0] === 'data') {
-      body += args[1];
+      if (body.length + args[1].length <= payloadSize) {
+        body += args[1];
```

Comment:
but if the initial body is small, and the second emit is huge, then you'll not store anything...